overlord/snapstate/snapmgr_test.go

```diff
@@ -842,12 +842,20 @@ func (s *snapmgrTestSuite) TestUpdateClassic(c *C) {
 	})
 
 	// snap installed with --classic, update needs classic, refresh without --classic works
-	_, err := snapstate.Update(s.state, "some-snap", "", snap.R(0), s.user.ID, snapstate.Flags{})
+	ts, err := snapstate.Update(s.state, "some-snap", "", snap.R(0), s.user.ID, snapstate.Flags{})
+	c.Assert(err, IsNil)
+	c.Assert(ts.Tasks(), Not(HasLen), 0)
+	snapsup, err := snapstate.TaskSnapSetup(ts.Tasks()[0])
 	c.Assert(err, IsNil)
+	c.Check(snapsup.Flags.Classic, Equals, true)
```

Comment:
to be clear I think such a test would show the problem @mvo5 hit, these were passing anyway